src/overlays/actors/ovl_En_Water_Effect/z_en_water_effect.c

```diff
@@ -28,30 +33,672 @@ const ActorInit En_Water_Effect_InitVars = {
+                        }
+                        ptr->unk_10.y = 0.0f;
+                        ptr->unk_1C.y = 0.0f;
+                        if (ptr->unk_2A == 0) {
+                            for (j = 0; j < 5; j++) {
+                                func_80A599E8(this, &ptr->unk_04, 2);
+                            }
+                            Audio_PlaySoundAtPosition(globalCtx, &ptr->unk_04, 30, NA_SE_EV_PLANT_BROKEN);
+                        }
+                        ptr->unk_2A++;
+                    }
+                }
+            }
+        }
+    }
+}
+
+void func_80A5A184(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnWaterEffect* this = THIS;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+    EnWaterEffectStruct* ptr = &this->unk_144[0];
+    u8 flag = false;
+    s16 i;
+
+    OPEN_DISPS(gfxCtx);
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    func_8012C2DC(globalCtx->state.gfxCtx);
+
+    for (i = 0; i < ARRAY_COUNT(this->unk_144); i++, ptr++) {
+        if (ptr->unk_00 == 4) {
+            if (!flag) {
+                gSPDisplayList(POLY_XLU_DISP++, object_water_effect_DL_004340);
+                gDPSetEnvColor(POLY_XLU_DISP++, 255, 10, 0, 0);
+                POLY_OPA_DISP = Gfx_SetFog(POLY_OPA_DISP, 255, 0, 0, 255, 500, 3600);
+                flag++;
+            }
+
+            gDPSetPrimColor(POLY_XLU_DISP++, 0x80, 0x80, (u8)ptr->unk_38, 0, 0, (u8)ptr->unk_3C);
+            gSPSegment(POLY_XLU_DISP++, 0x08,
+                       Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0, (ptr->unk_01 * -20) & 0x1FF,
+                                        0x20, 0x80));
+
+            Matrix_InsertTranslation(ptr->unk_04.x, ptr->unk_04.y, ptr->unk_04.z, MTXMODE_NEW);
+
+            if (ptr->unk_2A >= 2) {
+                Matrix_NormalizeXYZ(&globalCtx->billboardMtxF);
+            } else {
+                Matrix_RotateY(func_800DFC68(GET_ACTIVE_CAM(globalCtx)), MTXMODE_APPLY);
+            }
+
+            Matrix_Scale(ptr->unk_2C.x, ptr->unk_2C.y, 1.0f, MTXMODE_APPLY);
+
+            if ((i & 1) != 0) {
+                Matrix_InsertYRotation_f(M_PI, MTXMODE_APPLY);
+            }
+
+            gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                      G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_XLU_DISP++, object_water_effect_DL_0043E8);
+
+            if ((ptr->unk_2A & 1) == 0) {
+                Matrix_InsertTranslation(ptr->unk_04.x, ptr->unk_04.y + 5.0f, ptr->unk_04.z, MTXMODE_NEW);
+                Matrix_InsertXRotation_s(ptr->unk_28, MTXMODE_APPLY);
+                Matrix_Scale(ptr->unk_2C.z, ptr->unk_2C.z, ptr->unk_2C.z, MTXMODE_APPLY);
+
+                gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPDisplayList(POLY_OPA_DISP++, D_0406AB30);
```

Comment:
`D_0406AB30` is from gameplay keep. Should be able to include and use the symbol.